build_tools/azure/update_environments_and_lock_files.py

```diff
@@ -0,0 +1,388 @@
+"""Script to update CI environment files and associated lock files.
+
+To run it you need to be in the root folder of the scikit-learn repo:
+python build_tools/azure/update_environments_and_lock_files.py
+
+Two scenarii where this script can be useful:
```

Comment:
Nit: I think "scenarios" is the more commonly used term: https://www.merriam-webster.com/dictionary/scenario

```suggestion
Two scenarios where this script can be useful:
```